airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/load/task/DestinationTaskLauncherUTest.kt

```diff
@@ -0,0 +1,135 @@
+
+class DestinationTaskLauncherUTest {
+    private val taskScopeProvider: TaskScopeProvider<WrappedTask<ScopedTask>> =
+        mockk(relaxed = true)
+    private val catalog: DestinationCatalog = mockk(relaxed = true)
+    private val syncManager: SyncManager = mockk(relaxed = true)
+
+    // Internal Tasks
+    private val inputConsumerTask: InputConsumerTask = mockk(relaxed = true)
+    private val spillToDiskTaskFactory: SpillToDiskTaskFactory = mockk(relaxed = true)
+
+    // Implementor Tasks
+    private val setupTaskFactory: SetupTaskFactory = mockk(relaxed = true)
+    private val openStreamTaskFactory: OpenStreamTaskFactory = mockk(relaxed = true)
+    private val processRecordsTaskFactory: ProcessRecordsTaskFactory = mockk(relaxed = true)
+    private val processFileTaskFactory: ProcessFileTaskFactory = mockk(relaxed = true)
+    private val processBatchTaskFactory: ProcessBatchTaskFactory = mockk(relaxed = true)
+    private val closeStreamTaskFactory: CloseStreamTaskFactory = mockk(relaxed = true)
+    private val teardownTaskFactory: TeardownTaskFactory = mockk(relaxed = true)
+
+    // Checkpoint Tasks
+    private val flushCheckpointsTaskFactory: FlushCheckpointsTaskFactory = mockk(relaxed = true)
+    private val timedFlushTask: TimedForcedCheckpointFlushTask = mockk(relaxed = true)
+    private val updateCheckpointsTask: UpdateCheckpointsTask = mockk(relaxed = true)
+
+    // Exception handling
+    private val exceptionHandler: TaskExceptionHandler<LeveledTask, WrappedTask<ScopedTask>> =
+        mockk(relaxed = true)
+
+    private fun getDefaultDestinationTaskLauncher(
+        useFileTranfer: Boolean
+    ): DefaultDestinationTaskLauncher {
+        return DefaultDestinationTaskLauncher(
+            taskScopeProvider,
+            catalog,
+            syncManager,
+            inputConsumerTask,
+            spillToDiskTaskFactory,
+            setupTaskFactory,
+            openStreamTaskFactory,
+            processRecordsTaskFactory,
+            processFileTaskFactory,
+            processBatchTaskFactory,
+            closeStreamTaskFactory,
+            teardownTaskFactory,
+            flushCheckpointsTaskFactory,
+            timedFlushTask,
+            updateCheckpointsTask,
+            exceptionHandler,
+            useFileTranfer
+        )
+    }
+
+    // private val mainThread = newSingleThreadContext("UI thread")
+
+    @Test
+    fun `test that we don't start the spill-to-disk task when file transfer is enabled`() =
+        runTest {
+            coEvery { exceptionHandler.withExceptionHandling(any()) } returns mockk(relaxed = true)
+            coEvery { taskScopeProvider.launch(any()) } returns Unit
+
+            val destinationTaskLauncher = getDefaultDestinationTaskLauncher(true)
+            // This is needed to let the run method to complete
+            destinationTaskLauncher.handleTeardownComplete()
+            destinationTaskLauncher.run()
+
+            coVerify { catalog wasNot Called }
```

Comment:
Why verify this? It seems like an implementation detail.

Reply:
You're right, I should better make the catalog mock to return streams